server/schedule/checker/rule_checker.go

```diff
@@ -236,10 +236,14 @@ func (c *RuleChecker) fixLooseMatchPeer(region *core.RegionInfo, fit *placement.
 		checkerCounter.WithLabelValues("rule_checker", "fix-peer-role").Inc()
 		return operator.CreatePromoteLearnerOperator("fix-peer-role", c.cluster, region, peer)
 	}
+	if region.GetLeader() == nil {
+		checkerCounter.WithLabelValues("rule_checker", "region-no-leader").Inc()
+		return nil, nil
```

Comment:
Maybe we can return an error, so L158 can return

Reply:
agree

Reply:
If the region doesn't have a leader, should we check it?

Reply:
@rleungx  I think no need, address it.